lib/routes/token.js

```diff
@@ -89,6 +101,60 @@ function confirmRefreshToken(params) {
   });
 }
 
+function confirmJwt(params) {
+  var assertion = params.assertion;
+  logger.debug('jwt.confirm', assertion);
+
+  return JWTOOL.verify(assertion).catch(function(err) {
+    logger.info('jwt.invalid.verify', err.message);
+    throw AppError.invalidAssertion();
+  }).then(function(payload) {
+    logger.verbose('jwt.payload', payload);
+
+    var client = SERVICE_CLIENTS[payload.iss];
+
+    if (!client) {
+      logger.debug('jwt.client.unknown', payload.iss);
+      throw AppError.unknownClient(payload.iss);
+    }
+
+    var uid = payload.sub;
+    if (!uid || !(uid.length === 32 && HEX_STRING.test(uid))) {
+      logger.debug('jwt.invalid.sub', uid);
+      throw AppError.invalidAssertion();
+    }
+
+    if (payload.aud !== JWT_AUD) {
+      logger.debug('jwt.invalid.aud', payload.aud);
+      throw AppError.invalidAssertion();
+    }
+
+    if (!Scope(client.scope).has(payload.scope)) {
+      logger.debug('jwt.invalid.scopes', {
+        allowed: client.scope,
+        requested: payload.scope
+      });
+      throw AppError.invalidScopes();
+    }
+
+    var now = Date.now() / 1000;
+    if ((payload.iat || Infinity) > now) {
+      logger.debug('jwt.invalid.iat', { now: now, iat: payload.iat });
+      throw AppError.invalidAssertion();
+    }
+    if ((payload.exp || -Infinity) < now) {
```

Comment:
This does that, without introducing another branch (which jscs likes to complain about complexity). If there's no `payload.exp`, then `-Infinity` is used, and `-Infinity < now` will always be true, such that this would be considered expired.